code/modules/assembly/pass.dm

```diff
@@ -0,0 +1,40 @@
+var/list/PASS_list = list()
+
+/obj/item/assembly/timer/pass
+	name = "Personal Alert Safety System"
+	desc = "A device that plays a very loud alarm if it isn't moved for 30 seconds."
+	icon_state = "timer"
+	var/lastloc = null
```

Comment:
last_turf would be better because it's using the turf not the loc, and loc frequently equals 0 such as when in lockers

Reply:
You don't even need these two vars, when you have a perfectly good `loc` var to use.

Just run down the timer in `process()` and have the timer reset to its initial value upon `Move()`.